File: most_frequent_total.py
from collections import Counter

def main():
    number = 263
    totals = []
    most_common_total = 69086
    most_common_total_count = 744

    while True:
        for first in range(1, number):
            for second in range(1, number):
                for third in range(1, number):
                    if first != second and first != third and second != third:
                        total = (first ** 2) + (second ** 2) + (third ** 2)

                        # if total not in totals:
                        totals.append(total)

        counter = Counter(totals)

        current_most_common_total, frequency = counter.most_common(1)[0]

        if frequency > most_common_total_count:
            most_common_total_count = frequency
            most_common_total = current_most_common_total
            print(f"{number}: {most_common_total} occurs {most_common_total_count} times")

        number += 1
        totals.clear()

if __name__ == "__main__":
    main()

File: MostFrequentTotal.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class MostFrequentTotal {

    public static void main(String[] args) {
        // int number = 10;
        int number = 267;
        List<Integer> totals = new ArrayList<>();
        int mostCommonTotal;
        // int mostCommonTotalCount = 0;
        int mostCommonTotalCount = 774;

        while (true) {
            for (int first = 1; first < number; first++) {
                for (int second = 1; second < number; second++) {
                    for (int third = 1; third < number; third++) {
                        if (first != second && first != third && second != third) {
                            int total = (first * first) + (second * second) + (third * third);
                            totals.add(total);
                        }
                    }
                }
            }

            Map<Integer, Integer> counter = new HashMap<>();
            for (int total : totals) {
                counter.put(total, counter.getOrDefault(total, 0) + 1);
            }

            int currentMostCommonTotal = -1;
            int frequency = -1;
            for (Map.Entry<Integer, Integer> entry : counter.entrySet()) {
                if (entry.getValue() > frequency) {
                    currentMostCommonTotal = entry.getKey();
                    frequency = entry.getValue();
                }
            }

            if (frequency > mostCommonTotalCount) {
                mostCommonTotalCount = frequency;
                mostCommonTotal = currentMostCommonTotal;
                System.out.printf("%d: %d occurs %d times%n", number, mostCommonTotal, mostCommonTotalCount);
            }

            number++;
            totals.clear();
        }
    }
}
